app/controllers/web_hooks/slack_controller.rb

```diff
@@ -23,9 +10,50 @@ def build_create_response
     when 'url_verification'
       params[:challenge]
     when 'event_callback'
-      "Add Slack message event to db #{slack_message(params)}" if params[:event][:type] == 'message'
+      slack_event(params, params[:event][:type])
     else
       "Error"
     end
   end
+
+  protected def slack_event(event_callback, type)
+    if type == 'message'
+      "Add Slack message to db #{publish_event(Event.slack_messages.with_external_id(event_callback[:event_id]))}"
+    elsif type == 'app_mention'
+      "Add Slack announcement to db #{publish_announcement(event_callback)}"
+    else 'Error'
+    end
+  end
+
+  private def publish_event(event_scope)
+    event = event_scope.first_or_initialize
+    return unless event.new_record?
+
+    event.save!
+    Helios2Schema.subscriptions.trigger("eventPublished", {}, event)
+  end
+
+  private def publish_announcement(event_callback)
+    announcement = Announcement.create!(parse_announcement(event_callback))
+    Helios2Schema.subscriptions.trigger("announcementPublished", {}, announcement)
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  private def parse_announcement(event_callback)
+    # Ex: "guests: Welcome to Roy and Amanda from Under Armour on May 23rd at 11:00 am in Providence"
+    type, content = event_callback[:event][:text].split(/: /)
+
+    case type
+    when 'guests'
+      content = content.split(/ to | from | on | in /).map(&:strip)
```

Comment:
Would extracting this as a `handle_guests_command` method resolve the metrics issue above?